src/SFA.DAS.EmployerAccounts.Web/Authorization/DefaultAuthorizationHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using SFA.DAS.Authorization.Context;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using SFA.DAS.Authorization.Results;
+using System.Collections.Generic;
+using SFA.DAS.Authorization.Handlers;
+using static SFA.DAS.EmployerAccounts.Web.Authorization.ImpersonationAuthorizationContext;
+
+namespace SFA.DAS.EmployerAccounts.Web.Authorization
+{
+    public class DefaultAuthorizationHandler : IDefaultAuthorizationHandler
+    {
+        public Task<AuthorizationResult> GetAuthorizationResult(IReadOnlyCollection<string> options, IAuthorizationContext authorizationContext)
+        {
+            var authorizationResult = new AuthorizationResult();
+            authorizationContext.TryGet<Resource>("Resource", out var resource);
+            authorizationContext.TryGet<ClaimsIdentity>("ClaimsIdentity", out var claimsIdentity);
+            var resourceValue = resource != null ? resource.Value : "default";
+            var userRoleClaims = claimsIdentity?.Claims.Where(c => c.Type == claimsIdentity?.RoleClaimType);
+            if (userRoleClaims == null || userRoleClaims.All(claim => claim.Value != AuthorizationConstants.Tier2User))
+                return Task.FromResult(authorizationResult);
+
+            if (!CheckAllowedResourceList(resourceValue)) {
+                authorizationResult.AddError(new Tier2UserAccesNotGranted());
+            }
+            return Task.FromResult(authorizationResult);
+        }
+
+        public bool CheckAllowedResourceList(string resourceValue)
+        {
+            var resourceList = ResourceList.GetListOfAllowedResources();
+            return resourceList.Any(res => res == resourceValue);
+        }
+    }
+
+    public static class ResourceList
```

Comment:
> Should be provided by an inject able service so it can have different concrete implementations. e.g. IAuthorisationResourceRepository.Get() to return a List of Resources.
> 
> We were discussing other day, it would be nice to read from Configuration Table. For now we will keep this.

We are supposed to be improving the code base by following SOLID principals.  Good separation of concerns would indicate that this should be provided by a separate service/repository.